nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -299,12 +299,16 @@ sub GrubFs {
         copy $font, "$bootPath/converted-font.pf2" or die "cannot copy $font to $bootPath\n";
     }
     if ($splashImage) {
-        # FIXME: GRUB 1.97 doesn't resize the background image if it
-        # doesn't match the video resolution.
-        copy $splashImage, "$bootPath/background.png" or die "cannot copy $splashImage to $bootPath\n";
+        # Keeps the image's extension.
+        my ($filename, $dirs, $suffix) = fileparse($splashImage, qr"\..[^.]*$");
+        # The module for jpg is jpeg.
+        if ($suffix eq ".jpg") {
+            $suffix = ".jpeg";
+        }
+        copy $splashImage, "$bootPath/background$suffix" or die "cannot copy $splashImage to $bootPath\n";
         $conf .= "
-            insmod png
-            if background_image " . $grubBoot->path . "/background.png; then
+            insmod " . substr($suffix, 1) . "
```

Comment:
The previous behaviour:

 * Image did not load, text mode console

The new behaviour:

 * Exact same thing, image does not load, text mode console.

I'm guessing that an error message is quickly shown and hidden (`error: file '/grub/x86_64-efi/ext.mod`) as it would if I tried `insmod ext` in the console.